DQMToJSON: Add an argument to search by run.

schneiml opened the pull request:
The /data/json/samples endpoint only allows matching by dataset name so
far, and there was no other API way to search by run. This commit adds a
second regular expression argument `run` to search run numbers.

The performance impact of using a regular expression match vs. pure
integer operations needs to be evaluated on a larger volume of data;
however since the rx query on the dataset names performs quite well I
don't expect any problems.

Sample query:
http://.../dqm/online-dev/data/json/samples?run=^32177

schneiml committed:
Add a JSON endpoint to search by run.

The /data/json/samples endpoint only allows matching by dataset name so
far, and there was no other API way to search by run. This commit adds a
second regular expression argument `run` to search run numbers.

The performance impact of using a regular expression match vs. pure
integer operations needs to be evaluated on a larger volume of data;
however since the rx query on the dataset names performs quite well I
don't expect any problems.

Sample query:
http://.../dqm/online-dev/data/json/samples?run=^32177

rovere commented on src/cpp/DQM/serverext.cc:
```diff
@@ -2341,6 +2341,22 @@ class VisDQMToJSON
+	  }
+	}
+
 	for (size_t i = 0, e = samples.size(); i != e; ++i)
```
Ciao @schneiml , thanks for this.
Should we, maybe, embed the `if` logic inside so that we avoid almost 100% code duplication?
Also, what will happen if the use supply both filters?

schneiml replied:
Deduplicating the logic sure makes sense, though for the beginning I wanted to keep things simple. I can give it a try.

Using both filters does an _and_ which makes sense to me and works correctly.

schneiml committed:
Reorder code code to reduce duplication.

Also does only a single loop over the data now.

Includes some unrelated whitespace changes to remove the tabs from this
method.
